projects/observability/src/shared/dashboard/widgets/charts/cartesian-widget/cartesian-widget.model.ts

```diff
@@ -127,58 +139,174 @@ export class CartesianWidgetModel<TData> {
   @ModelInject(ColorService)
   private readonly colorService!: ColorService;
 
-  public getSeriesFetcher(): Observable<MetricSeriesFetcher<TData>> {
+  public getSeriesFetcher(): Observable<MetricSeriesFetcher<TSeriesInterval>> {
```

Comment:
Rather than having two separate fetchers, it may make more sense to have a single fetcher that returns both series and band data (so we don't have to parallel paths all the way up the stack). I would imagine something like

```typescript
export interface CartesianDataFetcher<TSeriesInterval> {
  getData(interval: TimeDuration): Observable<CombinedResults<TSeriesInterval>>;
  getRequestedInterval?(): TimeDuration | undefined;
}
```
This seems like it would greatly simplify the renderer (which basically has no changes if it's still working with a single fetcher), and the model would simplify too since it doesn't need to support the full fetcher building and interval resolution for both, but not sure if there's more issues here that's not coming through on read.